### PR TITLE
Remove duplicated code

### PR DESCRIPTION
**Pull request by gastaldi**
Reuses `split()` method in `ArtifactKey`

### CODE REVIEW

File: independent-projects/bootstrap/app-model/src/main/java/io/quarkus/maven/dependency/ArtifactKey.java [hunk 1]
```diff
@@ -34,4 +34,57 @@ default String toGacString() {
         }
         return buf.toString();
     }
+
+    static String[] split(String str, String[] parts, int fromIndex) {
```
Comment (aloubyansky) [inline independent-projects/bootstrap/app-model/src/main/java/io/quarkus/maven/dependency/ArtifactKey.java:38]: This method to me is an internal utility method, it doesn't feel like it belongs to the interface.

Comment (gastaldi) [reply independent-projects/bootstrap/app-model/src/main/java/io/quarkus/maven/dependency/ArtifactKey.java]: Right, I moved the method to an `ArtifactKeyUtil` class in the `io.quarkus.util` package

File: independent-projects/bootstrap/app-model/src/main/java/io/quarkus/maven/dependency/GACT.java [hunk 2]
```diff
@@ -8,59 +9,7 @@
     private static final long serialVersionUID = 2860156541775021365L;
 
     public static GACT fromString(String str) {
-        return new GACT(split(str, new String[4], str.length()));
-    }
-
-    protected static String[] split(String str, String[] parts, int fromIndex) {
```
Comment (aloubyansky) [inline independent-projects/bootstrap/app-model/src/main/java/io/quarkus/maven/dependency/GACT.java:14]: Maybe just keep this one and make it public? It's not great but I'd prefer it over creating `io/quarkus/util/ArtifactKeyUtil.java` for it. I think all the other flavors of the artifact API should go away eventually in favor of the `io.quarkus.maven.dependency` API.

Comment (gastaldi) [reply independent-projects/bootstrap/app-model/src/main/java/io/quarkus/maven/dependency/GACT.java:14]: As you wish :)

**Review by aloubyansky — APPROVED** [review]

**Pull request merged**